### PR TITLE
Added type definitions for listr module

### PR DESCRIPTION
**Pull request by durad**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/listr/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+declare class Listr {
+    constructor(tasks?: ReadonlyArray<ListrTask>, options?: ListrOptions);
+    constructor(options?: ListrOptions);
+    add(tasks: ListrTask | ReadonlyArray<ListrTask>): void;
+    run(ctx?: any): Promise<any>;
+}
+
+declare namespace Listr {
+}
```
Comment (plantain-00) [inline types/listr/index.d.ts]: It seems the empty namespace can be removed

**Review by plantain-00 — CHANGES_REQUESTED** [review]

File: types/listr/index.d.ts (hunk 1 above)
Comment (durad) [reply types/listr/index.d.ts]: Not really, if I remove empty namespace I get the following error when running `tsc`:

```
listr-tests.ts(1,24): error TS2497: Module '"DefinitelyTyped/types/listr/index"' 
resolves to a non-module entity and cannot be imported using this construct.
```

Comment (plantain-00) [reply types/listr/index.d.ts]: You can use `import Listr = require("listr");` instead.
the namespance is a hack and should be avoided

**Review by plantain-00 — APPROVED** [review]

File: types/listr/index.d.ts (hunk 1 above)
Comment (durad) [reply types/listr/index.d.ts]: I've exported other sub-types through namespace. This way they can be used as types in code.

**Comment by durad** [comment]
I've exported other sub-types through namespace. This way they can be used as types for variables in code.

**Pull request merged**